diff --git a/weave-cloud/apps/frontend/app.py b/weave-cloud/apps/frontend/app.py
index 26a8ae3..fa2912a 100644
--- a/weave-cloud/apps/frontend/app.py
+++ b/weave-cloud/apps/frontend/app.py
@@ -5,8 +5,9 @@ from flask import Flask, render_template, request
 from flask_opentracing import FlaskTracer
 from jaeger_client import Config
 import opentracing
+from sqlalchemy import exc
 
-from model import Star
+from model import db, Star
 
 
 __all__ = ["frontend_app"]
@@ -31,6 +32,15 @@ tracer = FlaskTracer(create_tracer)
 @tracer.trace("url_rule")
 def index():
     stars = Star.query.filter().all()
+    except exc.OperationalError:
+        cherrypy.log("Connection to database lost, trying a new connection")
+
+        # invalidate current pending query
+        db.session.rollback()
+
+        # will re-create a new connection pool
+        stars = Star.query.filter().all()
+
     return render_template('index.html', stars=stars)
 
 
diff --git a/weave-cloud/apps/frontend/model.py b/weave-cloud/apps/frontend/model.py
index 668709f..c8787eb 100644
--- a/weave-cloud/apps/frontend/model.py
+++ b/weave-cloud/apps/frontend/model.py
@@ -1,9 +1,14 @@
 # -*- coding: utf-8 -*-
-from flask_sqlalchemy import SQLAlchemy
+from flask_sqlalchemy import SQLAlchemy as SA
 
 __all__ = ["db", "Star"]
 
 
+class SQLAlchemy(SA):
+    def apply_pool_defaults(self, app, options):
+        SA.apply_pool_defaults(self, app, options)
+        options["pool_pre_ping"] = True
+
 db = SQLAlchemy()
 
 
